### PR TITLE
Improved support for in-memory databases

### PR DESCRIPTION
**Pull request by natemcmaster**
Support connection strings like `file:memdb1?mode=cached`

### CODE REVIEW

File: src/Microsoft.Data.Sqlite/SqliteCommand.cs [hunk 1]
```diff
@@ -122,6 +122,7 @@ public new virtual SqliteDataReader ExecuteReader(CommandBehavior behavior)
 
             do
             {
+                var isPragma = tail.StartsWith("PRAGMA ", StringComparison.OrdinalIgnoreCase);
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteCommand.cs]: Too brittle. Is this needed?

Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteCommand.cs]: `"PRAGMA journal_mode;"` throws with out this.

```
System.ArgumentNullException: Parameter name: SafeHandle cannot be null.
      Stack Trace:
           at System.StubHelpers.StubHelpers.SafeHandleAddRef(SafeHandle pHandle, Boolean& success)
           at Microsoft.Data.Sqlite.Interop.NativeMethods.sqlite3_column_type(Sqlite3StmtHandle stmt, Int32 iCol)
           at Microsoft.Data.Sqlite.SqliteDataReader.GetSqliteType(Int32 ordinal)
```

File: src/Microsoft.Data.Sqlite/SqliteConnection.cs [hunk 2]
```diff
@@ -94,12 +111,12 @@ public override void Open()
...
             }
 
-            var flags = Constants.SQLITE_OPEN_READWRITE | Constants.SQLITE_OPEN_CREATE;
+            var flags = Constants.SQLITE_OPEN_READWRITE | Constants.SQLITE_OPEN_CREATE | Constants.SQLITE_OPEN_URI;
             flags |= (ConnectionStringBuilder.Cache == SqliteConnectionCacheMode.Shared) ? Constants.SQLITE_OPEN_SHAREDCACHE : Constants.SQLITE_OPEN_PRIVATECACHE;
 
             var path = ConnectionStringBuilder.DataSource;
 
-            if (!path.Equals(":memory:", StringComparison.OrdinalIgnoreCase))
+            if (!IsInMemory(path))
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: I want to provide some better abstractions in this area. Connection strings that look like `Data Source=file:thedb?mode=memory; Cache=Shared;` are terrible.

File: src/Microsoft.Data.Sqlite/SqliteCommand.cs (hunk 1 above)
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteCommand.cs]: Ah, looks like we have another heuristic we can use. :smile:

File: src/Microsoft.Data.Sqlite/SqliteConnection.cs (hunk 2 above)
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: So, instead receive ConnectionStringBuilder params and translate to their equivalent `SQLITE_OPEN_*` [flags](https://www.sqlite.org/c3ref/c_open_autoproxy.html)?

File: src/Microsoft.Data.Sqlite/SqliteCommand.cs (hunk 1 above)
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteCommand.cs]: Updated to only match true "readonly" pragmas. Better?

**Comment by natemcmaster** [comment]
Ping. Even if we do add a first-class connection string parameter for "inmemory", we still want to support URI syntax, right?

**Pull request closed without merging**